training/src/anemoi/training/diagnostics/profilers.py

```diff
@@ -532,28 +532,33 @@ def get_speed_profiler_df(self, progressbar: _tqdm) -> pd.DataFrame:
         batch_size_tr = self.config.dataloader.batch_size.training
         batch_size_val = self.config.dataloader.batch_size.validation
 
-        training_rates_array = np.array(progressbar.training_rates)
-        speed_metrics["training_avg_throughput"] = training_rates_array.mean()
-        speed_metrics["training_avg_throughput_per_sample"] = training_rates_array.mean() / batch_size_tr
-
-        validation_rates_array = np.array(progressbar.validation_rates)
-        speed_metrics["validation_avg_throughput"] = validation_rates_array.mean()
-        speed_metrics["validation_avg_throughput_per_sample"] = validation_rates_array.mean() / batch_size_val
-
-        # Calculate per_sample metrics
+        num_iters_per_epoch = len(self.time_profiler.recorded_durations["run_training_batch"]) / len(
```

Comment:
everything here is asking `self.time_profiler.recorded_durations` to do something.
Should this code be moved to the object in `self.time_profiler`? 

This would allow, for instance, to have another object for self.time_profiler.
Or to disable time profiling, we would have a dummy self.time_profiler.

Reply:
After looking more closely, I think this code actually fails when self.time_profiler is the DummyProfiler.

Reply:
If the speed profiler is not dependant on the time profiler we could add an assert that the speed profiler shouldn't run then the time profiler is not enabled